test/core_test.js

```diff
@@ -0,0 +1,104 @@
+var expect = require('chai').expect;
+
+describe('Init', function() {
+  it("initializes the engine", function() {
+    var engine = require('../index');
+    engine.init({});
+    // TODO: Add test with actual options
+    expect(typeof engine).to.equal("object");
+  });
+});
+
+describe('Compile', function() {
+  it("creates a template object", function(done) {
+    var engine = require('../index');
+    engine.compile('{{baz}}', {}, function(err, tmpl) {
+      if (err) {
+        return done(err);
+      }
+      expect(typeof tmpl).to.equal("function");
+      expect(Object.keys(tmpl)).to.deep.equal(['_setup', '_child']);
+      expect(typeof tmpl._setup).to.equal('function');
+      expect(typeof tmpl._child).to.equal('function');
+      done();
+    });
+  });
+});
+
+describe('Render', function() {
+  it("renders a simple template", function(done) {
+    var engine = require('../index');
+    engine.compile('{{baz}}', {}, function(err, tmpl) {
+      if(err) {
+        return done(err);
+      }
+      engine.render(tmpl, {baz: 'bar'}, function(err,content) {
+        if(err) {
+          return done(err);
+        }
+        expect(content).to.equal('bar');
+        done();
+      });
+    });
+  });
+
+  // TODO: Fix this tests (the loading of helpers)
+  it.skip("renders a template with a helper", function(done) {
+    var engine = require('../index');
+    engine.init();
+    engine.compile('{{sum "[1,2,3,4,5]" 2}}', {}, function(err, tmpl) {
```

Comment:
@doowb @jonschlinkert for some reason this isn't working. The partials don't seem to register and I don't know why.